extensions/jfr/runtime/src/main/java/io/quarkus/jfr/runtime/runtime/JfrRuntimeBean.java

```diff
@@ -0,0 +1,96 @@
+package io.quarkus.jfr.runtime.runtime;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+
+import io.quarkus.jfr.runtime.config.JfrRuntimeConfig;
+import io.quarkus.runtime.ApplicationConfig;
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+import jdk.jfr.FlightRecorder;
+
+@ApplicationScoped
+public class JfrRuntimeBean {
+
+    @Inject
+    JfrRuntimeConfig jfrRuntimeConfig;
+
+    @Inject
+    QuarkusRuntimeInfo quarkusRuntimeInfo;
+
+    @Inject
+    ApplicationConfig applicationConfig;
+
+    private final Runnable runtimeEventTask = new RuntimeEventTask();
+    private final Runnable applicationEventTask = new ApplicationEventTask();
+    private final Runnable extensionEventTask = new ExtensionEventTask();
+
+    public void enable() {
+        FlightRecorder.register(QuarkusRuntimeEvent.class);
+        FlightRecorder.register(QuarkusApplicationEvent.class);
+        FlightRecorder.register(ExtensionEvent.class);
+        FlightRecorder.addPeriodicEvent(QuarkusRuntimeEvent.class, runtimeEventTask);
```

Comment:
Isn't this data static, in the sense that won't change during the life of the process? If so, shouldn't a single event on startup be enough?
Having a shutdown event would also look sweet. 

This also gave me an idea... We should be firing an event when we return the first REST response do we can accurately measure startup time until 1st response.